pageserver/src/tenant/timeline.rs

```diff
@@ -1924,6 +1936,20 @@ impl Timeline {
                 // to spawn_ondemand_logical_size_calculation.
                 let cancel = CancellationToken::new();
 
+                // in case we were created during pageserver initialization, wait for
+                // initialization to complete before proceeding. startup time init runs on the same
+                // runtime.
+                completion::Barrier::maybe_wait(self_clone.background_jobs_can_start.clone()).await;
```

Comment:
Took some looking, but of course this cannot be the same as the one used for *other* background jobs.

I will have to unroll some test config once I introduce another.

Reply:
Also forgot to actually switch off the init_done_rx usage...

Reply:
The next commit where I solved this I did also the refactoring which removed an argument instead of adding one, and then doing the refactoring. Please don't read commits up to that separatedly.